### PR TITLE
SOL-61 [FE] Show modal for first time user and pages on how to instal…

### PR DESCRIPTION
**Pull request by jasonchuasun**
…l MetaMast wallet

## Issue Link
- Backlog [Link](https://framgiaph.backlog.com/view/SPH_SOLIDITY-61)

## Target Page | Component | Function
- `/help`

## Pre-condition
- Add here

## Steps | Commands
1. With no user currently signed in and MetaMask extension not setup, click on "Connect wallet"

## Expected Output
- The modal should appear and the button should redirect to `/help` page with steps on how to setup MetaMask wallet extension.

## Definition of Done
- redirects to `/help` page should be working
- site links should be working
- help button beside search button should be working
- highlighted links (i.e. Metamask extension link, Sepolia link , Alchemy links) should be working and open a new tab

## Test View Points
- [ ] Add here

## Notes
N/A

## Screenshots | Recordings
[SUNBLOCK.webm](https://github.com/framgia/sph-solidity/assets/118784390/191339d9-a625-4311-b208-3bc51cef7efe)

### General
- [ ] Does the code meet the requirements of the project?
- [ ] Is the code readable and maintainable?
- [ ] Are there any unnecessary or redundant lines of code?
- [ ] Are there any long methods or functions that should be refactored?
- [ ] Are there any performance issues?

### Functionality
- [ ] Are there any logical errors?
- [ ] Are edge cases handled properly?
- [ ] Are there any potential security vulnerabilities?

### Code Style
- [ ] Does the code adhere to the project's coding standards?
- [ ] Are variable and function names descriptive and meaningful?
- [ ] Is there proper use of comments and documentation?
- [ ] Is the code properly formatted? Are indentation, spacing, and formatting consistent?

### Security
- [ ] Is the website secure from common attacks?
- [ ] Are inputs properly validated and sanitized?
- [ ] Are sensitive data and credentials properly protected?

### Testing
- [ ] Are there sufficient test cases?
- [ ] Do the test cases cover all functionality?
- [ ] Are the test cases properly written and documented?

### Additional Comments
- [ ] Are there any other issues or concerns with the code?
- [ ] Are there any suggestions for improvement?
- [ ] Are there any positive aspects of the code that should be highlighted?

### CODE REVIEW

File: fe/src/components/global/navbar.vue [hunk 1]
```diff
@@ -2,12 +2,23 @@
...
         <WalletButton />
       </div>
     </nav>
   </header>
 </template>
+<script setup lang="ts">
+const router = useRouter();
 
-<script setup lang="ts"></script>
+const handleHelp = () => {
```
Comment (kenorenzsun) [inline fe/src/components/global/navbar.vue:21]:
```suggestion
const handleHelp = (): void => {
```

File: fe/src/components/global/wallet-button.vue [hunk 2]
```diff
@@ -47,7 +47,7 @@ const { middleTruncate } = useUtils();
 
 const isDropdownOpen = ref<boolean>(false);
 const useWallet = useWalletStore();
-const { updateStatus } = useWallet;
+const { updateStatus, updateIsShowModal } = useWallet;
 const { isConnected, address } = storeToRefs(useWallet);
 
 const connectWallet = async () => {
```
Comment (kenorenzsun) [inline fe/src/components/global/wallet-button.vue]:
```suggestion
const connectWallet = async (): Promise<void> => {
```

File: fe/src/pages/create-campaign.vue [hunk 3]
```diff
@@ -138,6 +151,11 @@ const { value: date } = useField<CreateCampaignRequest["date"]>("date");
 const isLoading = ref<boolean>(false);
 const useWallet = useWalletStore();
 const { isConnected } = storeToRefs(useWallet);
+const { updateIsShowModal } = useWallet;
+
+const handleCloseModal = () => {
```
Comment (kenorenzsun) [inline fe/src/pages/create-campaign.vue]:
```suggestion
const handleCloseModal = (): void => {
```

File: fe/src/pages/create-campaign.vue [hunk 4]
```diff
@@ -138,6 +151,11 @@ const { value: date } = useField<CreateCampaignRequest["date"]>("date");
...
+  updateIsShowModal(false);
+};
 
 const warning = () => {
```
Comment (kenorenzsun) [inline fe/src/pages/create-campaign.vue]:
```suggestion
const warning = (): void => {
```

File: fe/src/pages/help.vue [hunk 5]
```diff
@@ -0,0 +1,171 @@
...
+        >
+      </div>
+    </div>
+  </div>
+</template>
+<script setup>
+const router = useRouter();
+const activeStep = ref(1);
+const totalSteps = ref(6);
+const handleNext = () => {
```
Comment (kenorenzsun) [inline fe/src/pages/help.vue]:
```suggestion
const handleNext = () : void => {
```

File: fe/src/pages/help.vue [hunk 6]
```diff
@@ -0,0 +1,171 @@
...
+          >Go to dashboard</BaseButton
+        >
+      </div>
+    </div>
+  </div>
+</template>
+<script setup>
+const router = useRouter();
+const activeStep = ref(1);
+const totalSteps = ref(6);
```
Comment (kenorenzsun) [inline fe/src/pages/help.vue:153]:
```suggestion
const activeStep = ref<number>(1);
const totalSteps = ref<number>(6);
```

File: fe/src/pages/help.vue [hunk 7]
```diff
@@ -0,0 +1,171 @@
...
+
+const handlePrevious = () => {
+  if (activeStep.value > 1) {
+    activeStep.value = activeStep.value - 1;
+  }
+};
+
+const handleBack = () => {
+  router.back();
+};
```
Comment (kenorenzsun) [inline fe/src/pages/help.vue]:
```suggestion
const handlePrevious = (): void => {
  if (activeStep.value > 1) {
    activeStep.value = activeStep.value - 1;
  }
};

const handleBack = (): void => {
  router.back();
};
```

File: fe/src/pages/index.vue [hunk 8]
```diff
@@ -67,11 +80,19 @@
...
+import { useWalletStore } from "~/store/wallet";
 
 const itemsPerPage = ref(5);
 const currentPage = ref(1);
 const cardValueSample = ref<CampaignCardProps>(CardSample);
 
+const walletStore = useWalletStore();
+const { updateIsShowModal } = walletStore;
+
+const handleCloseModal = () => {
```
Comment (kenorenzsun) [inline fe/src/pages/index.vue]:
```suggestion
const handleCloseModal = (): void=> {
```

File: fe/src/store/wallet.ts [hunk 9]
```diff
@@ -4,13 +4,15 @@ export interface WalletStore {
...
 
 export const useWalletStore = defineStore("walletStore", {
   state: (): WalletStore => ({
     isConnected: false,
     address: "",
     balance: 0,
+    isShowModal: false,
   }),
   actions: {
     updateStatus(status: boolean, address: string) {
```
Comment (kenorenzsun) [inline fe/src/store/wallet.ts:18]:
```suggestion
    updateStatus(status: boolean, address: string): void {
```

File: fe/src/store/wallet.ts [hunk 10]
```diff
@@ -20,5 +22,8 @@ export const useWalletStore = defineStore("walletStore", {
     updateBalance(balance: number) {
```
Comment (kenorenzsun) [inline fe/src/store/wallet.ts:22]:
```suggestion
    updateBalance(balance: number): void {
```

File: fe/src/store/wallet.ts [hunk 11]
```diff
@@ -20,5 +22,8 @@ export const useWalletStore = defineStore("walletStore", {
...
       this.balance = balance;
     },
+    updateIsShowModal(status: boolean) {
```
Comment (kenorenzsun) [inline fe/src/store/wallet.ts:25]:
```suggestion
    updateIsShowModal(status: boolean): void {
```

**Review by kenorenzsun — DISMISSED** [review]

**Review by kenorenzsun — DISMISSED** [review]

**Comment by ipilikent** [comment]
Comment
• Problem
- The buttons are not aligned
image

• Expected Output
- The buttons should be aligned
image
image

**Comment by ipilikent** [comment]
## Approved PR
-  All displayed the same as the Figma design
- Can go back landing page
- Can click question mark icon
- Can display modal 
- Can close the modal
- All buttons are working 
- Can redirect MetaMask wallet setup
- Can redirect to the MetaMask extension
- Can redirect tot the Sepolia Faucet and Alchemy website
- The validation messages are working and displayed
- Scrollable

## Proof
<details>

Sol- 61 FE

</details>

Test Case: https://docs.google.com/spreadsheets/d/114GrEn-LQQ8saIvjz9zqenuk05qmO4dN/edit#gid=1530463528
-----
- Total Test time:  0.75 hr.
- Total Comments: 1
- Total Iterations: 0

**Review by ipilikent — APPROVED** [review]

File: fe/src/store/wallet.ts [hunk 12]
```diff
@@ -4,21 +4,26 @@ export interface WalletStore {
...
       this.isConnected = status;
       this.address = address;
     },
-    updateBalance(balance: number) {
+    updateBalance(balance: number): void {
       this.balance = balance;
     },
+    updateIsShowModal(status: boolean): void {
+      this.isShowModal = status;
+    },
```
Comment (chevyB) [inline fe/src/store/wallet.ts]: I think no need for return type for the actions

File: fe/src/pages/index.vue [hunk 13]
```diff
@@ -67,11 +80,19 @@
...
 const itemsPerPage = ref(5);
 const currentPage = ref(1);
 const cardValueSample = ref<CampaignCardProps>(CardSample);
 
+const walletStore = useWalletStore();
+const { updateIsShowModal } = walletStore;
+
+const handleCloseModal = (): void => {
+  updateIsShowModal(false);
+};
```
Comment (chevyB) [inline fe/src/pages/index.vue]:
```suggestion
const handleCloseModal = () =>  updateIsShowModal(false)
```

**Review by chevyB — DISMISSED** [review]

**Review by kenorenzsun — DISMISSED** [review]

**Review by ipilikent — APPROVED** [review]

File: fe/src/assets/css/tailwind.css [hunk 14]
```diff
@@ -32,4 +32,4 @@ body {
   .dropdown-item {
     @apply p-2 text-left hover:bg-gray-200;
   }
-}
+}
```
Comment (johnpaul-sun) [inline fe/src/assets/css/tailwind.css]: please add a new line at the bottom 🙇‍♂️

**Review by johnpaul-sun — APPROVED** [review]
Approved ✅

Review Time: 10 mins
Comments: 1
IT: 0

**Review by ipilikent — APPROVED** [review]

**Review by kenorenzsun — APPROVED** [review]

**Pull request merged**